r2/r2/lib/providers/search/cloudsearch.py

```diff
@@ -1091,3 +797,38 @@ def customize_query(self, bq=u''):
         if not self.include_over18:
             queries.append('over18:0')
         return self.create_boolean_query(queries)
+
+
+class CloudSearchProvider(SearchProvider):
+    '''Provider implementation: wrap it all up as a SearchProvider'''
+
+    InvalidQuery = (InvalidQuery,)
+    SearchException = (SearchHTTPError,)
+
+    SearchQuery = LinkSearchQuery
+
+    SubredditSearchQuery = CloudSearchSubredditSearchQuery
+
+    sorts = LinkSearchQuery.sorts_menu_mapping
+
+    def run_changed(drain=False, min_size=int(getattr(g, 'SOLR_MIN_BATCH', 500)), limit=1000, sleep_time=10, 
+            use_safe_get=False, verbose=False):
```

Comment:
this method (and its partner in solrsearch) also needs to take `self` as its first argument.